.github/workflows/build.yml

```
@@ -33,28 +33,34 @@ jobs:
     - name: Checkout
       uses: actions/checkout@v4
 
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v5
+    - name: Build tfx-bsl
+      id: build-tfx-bsl
+      uses: ./.github/reusable-build
       with:
         python-version: ${{ matrix.python-version }}
 
-    - name: Upgrade pip
-      run: |
-        python -m pip install --upgrade pip
 
-    - name: Build the manylinux2010 image
-      run: docker compose build manylinux2010
-
-    - name: Build the package for Python ${{ matrix.python-version }}
-      run: |
-        version="${{ matrix.python-version }}"
-        docker compose run -e PYTHON_VERSION=$(echo "$version" | sed 's/\.//') manylinux2010
-
-    - name: Upload wheel artifact for Python ${{ matrix.python-version }}
-      uses: actions/upload-artifact@v3
-      with:
-        name: tfx-bsl-wheel-py${{ matrix.python-version }}
-        path: dist/*.whl
+  upload_to_pypi:
+    name: Upload to PyPI
+    runs-on: ubuntu-latest
+    if: (github.event_name == 'release' && startsWith(github.ref, 'refs/tags')) || (github.event_name == 'workflow_dispatch')
+    needs: [build]
+    environment:
+      name: pypi
+      url: https://pypi.org/p/struct2tensor/
+    permissions:
+      id-token: write
```

I know this PR isn't focused on builds, but we need to modify the build workflow anyway, so I'll mention this here:

It would be good to add [attestations](https://github.blog/changelog/2024-06-25-artifact-attestations-is-generally-available/) to establish SLSA provenance for our artifacts:

```suggestion
      id-token: write
      attestations: write
```